docs/modules/ROOT/pages/guide/private-network.adoc

```diff
@@ -0,0 +1,105 @@
++
+```
+cast send --rpc-url NETWORK_RPC --private-key PRIVATE_KEY --create 0x604580600e600039806000f350fe7fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffe03601600081602082378035828234f58015156039578182fd5b8082525050506014600cf3
+```
+
+. Replace the `deterministicDeployment` function in `hardhat.config.ts` with the following code, replacing `YOUR_CONTRACT_ADDRESS` and `YOUR_WALLET_ADDRESS`:
++
+[source,jsx]
+----
+const deterministicDeployment = (): DeterministicDeploymentInfo => {
+    return {
+        factory: "YOUR_CONTRACT_ADDRESS",
+        deployer: "YOUR_WALLET_ADDRESS",
+        funding: BigNumber.from(100000).mul(BigNumber.from(100000000000)).toString(),
+        signedTx: "0xf8a58085174876e800830186a08080b853604580600e600039806000f350fe7fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffe03601600081602082378035828234f58015156039578182fd5b8082525050506014600cf326a0b1fd9f4102283a663738983f1aac789e979e220a1b649faa74033f507b911af5a061dd0f2f6f2341ee95913cf94b3b8a49cac9fdd7be6310da7acd7a96e31958d7",
+    };
+};
+----
+
+. Run the following command to deploy the Safe contracts (don't worry about the verification errors):
++ 
+```
+npm run deploy-all custom
+```
+
+. Navigate to the https://defender.openzeppelin.com/v2/#/manage/networks/private[`Private Networks` page on Defender, window=_blank] and click on the edit button of the network you created. 
++
+image::guide-edit-private-network.png[Edit Private Network on Defender]
+
+. Copy the following addreses from the Safe contracts deployment output and paste them on Defender:
+* Safe Master Address: `Safe`
+* Safe Proxy Factory Address: `SafeProxyFactory`
+* Safe Multi-Send Call-Only Address: `MultiSendCallOnly`
+* Safe Create Call Address: `CreateCall`
+
+[[create-subgraph]]
+== 3. Create subgraph
+
+Subgraphs on Defender are powered by https://thegraph.com[TheGraph, window=_blank]. In order to create one for a Private Network, the network must be first supported by the TheGraph. https://github.com/graphprotocol/graph-tooling/blob/121843e982c69ffb31aae911431a68a2349ea062/packages/cli/src/protocols/index.ts#L91[Here's, window=_blank] the list of supported networks as data sources. Follow these steps to create a subgraph:
+
+. Clone the Defender subgraph toolkit repository and install the dependencies:
++
+```
+git clone https://github.com/OpenZeppelin/defender-subgraphs && cd defender-subgraphs && yarn
+```
+
+. Follow the steps in the https://github.com/OpenZeppelin/defender-subgraphs/blob/main/README.md[README,window=_blank].
+. Copy the subgraph URL and paste it in the `Subgraph URL` field on the Defender Private Network configuration.
+
+image::guide-subgraph-private-network.png[Subgraph URL on Defender]
+
+[[next-steps]]
+== Next steps
+
+Congratulations! You have successfully created a complete Private Network. You can now use it to deploy and test your smart contracts with the Safe contracts and subgraph.
```

Comment:
Nit: Should we word it as "you have successfully added a private network" i think "created a complete private network" sounds like we're providing the infrastructure. Same for the title.